packages/@react-stately/utils/src/number.ts

```diff
@@ -20,9 +20,22 @@ export function clamp(value: number, min: number = -Infinity, max: number = Infi
 
 export function roundToStepPrecision(value: number, step: number): number {
   let roundedValue = value;
+  let precision = 0;
   let stepString = step.toString();
-  let pointIndex = stepString.indexOf('.');
-  let precision = pointIndex >= 0 ? stepString.length - pointIndex : 0;
+  // Handle negative exponents in exponential notation (e.g., "1e-7" → precision 8)
+  let eIndex = stepString.toLowerCase().indexOf('e-');
+  if (eIndex > 0) {
+    let significand = stepString.slice(0, eIndex);
+    let exponent = stepString.slice(eIndex + 1);
+    let pointIndex = significand.indexOf('.');
+    let decimals = pointIndex >= 0 ? significand.length - pointIndex : 0;
+    precision = Math.abs(Number(exponent)) + decimals + 1;
+  } else {
```

Comment:
I think this is a little more succint and explains where the `+ 1` was coming from
```suggestion
  if (eIndex > 0) {
    precision = Math.abs(Math.floor(Math.log10(Math.abs(step)))) + eIndex;
  } else {
```

Reply:
though don't know if it's susceptible to floating point precision issues... i haven't thought of a case yet